linera-core/src/unit_tests/worker_tests.rs

```diff
@@ -355,35 +356,12 @@ fn update_recipient_direct(recipient: ChainId, certificate: &Certificate) -> Cro
     }
 }
 
-#[test(tokio::test)]
-async fn test_memory_handle_block_proposal_bad_signature() {
-    let storage = MemoryStorage::make_test_storage(None).await;
-    run_test_handle_block_proposal_bad_signature(storage).await;
-}
-
-#[cfg(feature = "rocksdb")]
-#[test(tokio::test)]
-async fn test_rocks_db_handle_block_proposal_bad_signature() {
-    let _lock = ROCKS_DB_SEMAPHORE.acquire().await;
-    let (storage, _dir) = RocksDbStorage::make_test_storage(None).await;
-    run_test_handle_block_proposal_bad_signature(storage).await;
-}
-
-#[cfg(feature = "aws")]
-#[test(tokio::test)]
-async fn test_dynamo_db_handle_block_proposal_bad_signature() {
-    let storage = DynamoDbStorage::make_test_storage(None).await;
-    run_test_handle_block_proposal_bad_signature(storage).await;
-}
-
-#[cfg(feature = "scylladb")]
-#[test(tokio::test)]
-async fn test_scylla_db_handle_block_proposal_bad_signature() {
-    let storage = ScyllaDbStorage::make_test_storage(None).await;
-    run_test_handle_block_proposal_bad_signature(storage).await;
-}
-
-async fn run_test_handle_block_proposal_bad_signature<S>(storage: S)
+#[test_case((), MemoryStorage::make_test_storage(None).await; "memory")]
+#[cfg_attr(feature = "rocksdb", test_case(ROCKS_DB_SEMAPHORE.acquire().await, RocksDbStorage::make_test_storage(None).await.0; "rocksdb"))]
```

Comment:
I think this drops the `_dir`, and may delete it :thinking: We might need to write a wrapper type to keep the semaphore permit and the the directory guard.

Reply:
Good catch!

And I realized that `client_tests` could use `test_case`, too, so I'll add that as well.